src/uk/gov/defra/ffc/DefraUtils.groovy

```diff
@@ -47,14 +47,47 @@ def String __generateTerraformInputVariables(Map inputs) {
     return __mapToString(inputs).join("\n")
 }
 
+def destroyInfrastructure(repo_name, pr_code) {
+  sshagent(['helm-chart-creds']) {
+    echo "destroyInfrastructure"
+    dir('terragrunt') {
+      // git clone repo...
+      git credentialsId: 'helm-chart-creds', url: 'git@gitlab.ffc.aws-int.defra.cloud:terraform_sqs_pipelines/terragrunt_sqs_queues.git'
+      dir("london/eu-west-2/ffc") {
+        def dirName = "${repo_name}-pr${pr_code}-*"
+        echo "finding previous var files in directories matching ${dirName}";
+        def varFiles = findFiles glob: "${dirName}/vars.tfvars";
+        echo "found ${varFiles.size()} directories to tear down";
+        for (varFile in varFiles) {
+          def path = varFile.getPath().substring(0, varFile.getPath().lastIndexOf("/"))
+          echo "running terragrunt in ${path}"
+          dir(path) {
+            // terragrunt destroy
+            sh("terragrunt destroy -var-file='${varFile.getName()}' -auto-approve")
+          }
+          // delete the pr dir
+          echo "removing from git"
+          sh "git rm -fr ${path}"
+        }
+        // commit the changes back
+        echo "persisting changes in repo"
+        sh "git commit -m \"Removing infrastructure created for ${repo_name}#${pr_code}\" ; git push --set-upstream origin master"
```

Comment:
Only run the `git commit` if `varFiles.size() > 0`. If you run the `destroyInfrastructure` function when there is nothing to destroy the pipeline fails as there git complains there is nothing to commit.